### PR TITLE
Enable QueryStore tab to the database properties

### LINKED ISSUE #2212: Provide SMO database properties experience to the QueryStore Tab

### PR DESCRIPTION
**Pull request by ssreerama**
This PR adds the ability to view/update query store options to the database proeprties.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs [hunk 1]
```diff
@@ -89,6 +98,33 @@ static DatabaseHandler()
             displayFileTypes.TryAdd(FileType.Log, SR.prototype_file_logFile);
             displayFileTypes.TryAdd(FileType.FileStream, SR.prototype_file_filestreamFile);
 
+            displayOperationModeOptions.TryAdd(QueryStoreOperationMode.Off, CommonConstants.QueryStoreOperationMode_Off);
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:101]: I don't think we should be localizing option values like OFF and READ_ONLY - do we do that in SSMS?

@erinstellato-ms what do you think?

File: src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype130.cs [hunk 2]
```diff
@@ -66,6 +82,64 @@ protected override void SaveProperties(Database db)
...
+            if (db.IsSupportedObject<QueryStoreOptions>())
+            {
+                try
+                {
+                    // Querystore propeties do not initialized at the time of create database
+                    var desiredState = this.currentState.queryStoreOptions.DesiredState;
+                }
+                catch (Exception ex)
+                {
+                    areQueryStorePropertiesInitialized = false;
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype130.cs:95]: What exception are you expecting to be thrown here? I'm not sure what "initialized" would mean in this case, or why we'd want to skip setting the values completely in that case.

**Review by corivera — APPROVED** [review]
Looks good to me pending other people's comments.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs (hunk 1 above)
Comment (ssreerama) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:101]: Other couple of dropdown options(All,Auto,None,Custom) are translated in the MS quey store help doc, so localized these too.

Comment (erinstellato-ms) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:101]: What's our general practice on localizing options in ADS?  Specifically, options in a drop down?  Example: Backup functionality (in preview), we have Set backup compression as a drop down with three options.  Are those localized?

Fundamentally...first, let's be consistent across ADS.  Second, I will have to install another language of SSMS to see what we do there :)

File: src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype130.cs (hunk 2 above)
Comment (ssreerama) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype130.cs:95]: While create database the value this.currentState.queryStoreOptions is not null but its properties (ex: DesiredState...etc) throws "'this.currentState.queryStoreOptions.DesiredState' threw an exception of type 'System.NullReferenceException'"

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs (hunk 1 above)
Comment (ssreerama) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:101]: I have created a task to verify and add/remove all such dropdown options:
https://github.com/microsoft/sqltoolsservice/issues/2213

**Review by erinstellato-ms — APPROVED** [review]

**Review by corivera — APPROVED** [review]
Approved pending my other comments

File: src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs [hunk 3]
```diff
@@ -562,6 +564,21 @@ public DatabaseData(CDataContainer context, string databaseName)
...
+                // Supported from Sql Server 2016 and higher
+                // QueryStore properties are not initialized at the time of database creation
+                if (db.IsSupportedObject<QueryStoreOptions>())
+                {
+                    try
+                    {
+                        var queryStoreActualStateProperty = db.QueryStoreOptions.ActualState;
+                        this.queryStoreOptions = db.QueryStoreOptions;
+                    }
+                    catch (NullReferenceException)
```
[the remark this answers is not in the document]
Comment (ssreerama) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs:576]: db.QueryStoreOptions is not null, but the properties(actualState ...) inside it are Not initialized and when accessing them it throws the null reference exception.

[the remark this answers is not in the document]
Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs:576]: Could you add a comment here for that?

**Pull request merged**